org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/cleanup/CleanUpUtils.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.cleanup;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.manipulation.CleanUpContextCore;
+import org.eclipse.jdt.core.manipulation.ICleanUpFixCore;
+import org.eclipse.jdt.core.refactoring.CompilationUnitChange;
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+import org.eclipse.jdt.ls.core.internal.TextEditConverter;
+import org.eclipse.jdt.ls.core.internal.corrections.ASTResolving;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.eclipse.lsp4j.TextEdit;
+
+/**
+ * Functions for working with JDT ICleanUpCore and ISimpleCleanUp.
+ *
+ * @author datho7561
+ */
+public class CleanUpUtils {
+
+	private CleanUpUtils() {
+	}
+
+	/**
+	 * Returns the clean up context for the given text document.
+	 *
+	 * @param textDocumentId
+	 *            the text document to get the clean up context for
+	 * @param monitor
+	 *            the progress monitor
+	 * @return the clean up context for the given text document
+	 */
+	public static CleanUpContextCore getCleanUpContext(TextDocumentIdentifier textDocumentId, IProgressMonitor monitor) {
+		ICompilationUnit unit = JDTUtils.resolveCompilationUnit(textDocumentId.getUri());
+		CompilationUnit ast = ASTResolving.createQuickFixAST(unit, monitor);
```

Comment:
Didn't know about this helper method for getting an AST. There's probably a bunch in JDT. I noticed that there's `ASTParser.setCompilerOptions(Map<String, String>)` . It seems to control the problems that will be flagged according to the javadoc. Could this be used to hide any potentially unwanted side-effects (eg. flickering the options on the UI side when calling `project.setOptions(..)`) from the user ?

I'd still ready the javadoc for it but might be worth investigating.

Reply:
Yep. That appears to work! Thanks!